src/storage-adapters/index.ts

```diff
@@ -0,0 +1,23 @@
+import { StorageAdapter, StorageRegistry } from './types';
+
+export * from './types';
+const storageRegistry: StorageRegistry = {};
+
+export function registerStorageAdapter(
+	name: string,
+	adapter: StorageAdapter,
+): void {
+	storageRegistry[name] = adapter;
+}
+
+export function getStorageAdapter(name: string): StorageAdapter {
```

Comment:
I wonder why typescript does not complain about the return type as `return storageRegistry[name]` can return undefined.

Reply:
I think this is because of the `const storageRegistry` returned which is declared on line 4.

Reply:
@nitishagar from my understanding the StorageRegistry type is a dictionary that can be empty. Hence, storageRegistry[name] can return undefined when the key is not existing, right? Sorry if I miss something here

Reply:
@fisehara it's a known/long time typescript thing, you can enable it by using [`noUncheckedIndexedAccess`](https://www.typescriptlang.org/tsconfig#noUncheckedIndexedAccess)

Reply:
Thanks @Page- , I've added the flag and updated the function definition to:
`export function getStorageAdapter(name: string): StorageAdapter | undefined {`

Reply:
The storage adapter concept was removed now that we're creating the S3 objects from pinejs